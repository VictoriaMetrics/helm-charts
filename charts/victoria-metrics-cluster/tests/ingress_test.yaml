suite: ingress templates test
chart:
  version: 0.1.1
  appVersion: 0.1.0
templates:
  - ingress.yaml
tests:
  - it: ingress should match snapshot
    set:
      vmauth:
        enabled: true
        ingress:
          enabled: true
          annotations:
            kubernetes.io/ingress.class: nginx
          hosts:
            - name: vmselect.node.local
              path: /select/test
              port: http
      vminsert:
        ingress:
          enabled: true
          annotations: 
            kubernetes.io/ingress.class: alb
          tls:
            - secretName: insert-ingress-tls
              hosts:
                - insert.local
      vmselect:
        ingress:
          enabled: true
          annotations: 
            kubernetes.io/ingress.class: kong
    asserts:
      - matchSnapshot: {}
  - it: ingress should match snapshot with fullnameOverride and extraLabels
    set:
      vminsert:
        name: insert
        fullnameOverride: vminsert-node
        extraLabels:
          extraLabelName: insertExtraLabelValue
        ingress:
          extraLabels:
            extraIngressLabelName: insertExtraIngressLabelValue
      vmauth:
        enabled: true
        name: auth
        fullnameOverride: vmauth-node
        extraLabels:
          extraLabelName: authExtraLabelValue
        ingress:
          extraLabels:
            extraIngressLabelName: authExtraIngressLabelValue
      vmselect:
        name: select
        fullnameOverride: vmselect-node
        extraLabels:
          extraLabelName: selectExtraLabelValue
        ingress:
          extraLabels:
            extraIngressLabelName: selectExtraIngressLabelValue
      vmstorage:
        name: storage
        fullnameOverride: vmstorage-node
        extraLabels:
          extraLabelName: storageExtraLabelValue
        ingress:
          extraLabels:
            extraIngressLabelName: storageExtraIngressLabelValue
    asserts:
      - matchSnapshot: {}
