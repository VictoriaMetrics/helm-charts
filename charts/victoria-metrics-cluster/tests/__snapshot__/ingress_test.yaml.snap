ingress should match snapshot:
  1: |
    apiVersion: networking.k8s.io/v1
    kind: Ingress
    metadata:
      annotations:
        kubernetes.io/ingress.class: nginx
      labels:
        app: vmauth
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: victoria-metrics-cluster
        app.kubernetes.io/version: 0.1.0
        helm.sh/chart: victoria-metrics-cluster-0.1.1
      name: RELEASE-NAME-victoria-metrics-cluster-vmauth
      namespace: NAMESPACE
    spec:
      rules:
        - host: vmselect.node.local
          http:
            paths:
              - backend:
                  service:
                    name: RELEASE-NAME-victoria-metrics-cluster-vmauth
                    port:
                      name: http
                path: /select/test
                pathType: Prefix
  2: |
    apiVersion: networking.k8s.io/v1
    kind: Ingress
    metadata:
      annotations:
        kubernetes.io/ingress.class: alb
      labels:
        app: vminsert
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: victoria-metrics-cluster
        app.kubernetes.io/version: 0.1.0-cluster
        helm.sh/chart: victoria-metrics-cluster-0.1.1
      name: RELEASE-NAME-victoria-metrics-cluster-vminsert
      namespace: NAMESPACE
    spec:
      rules:
        - host: vminsert.local
          http:
            paths:
              - backend:
                  service:
                    name: RELEASE-NAME-victoria-metrics-cluster-vminsert
                    port:
                      name: http
                path: /insert
                pathType: Prefix
      tls:
        - hosts:
            - insert.local
          secretName: insert-ingress-tls
  3: |
    apiVersion: networking.k8s.io/v1
    kind: Ingress
    metadata:
      annotations:
        kubernetes.io/ingress.class: kong
      labels:
        app: vmselect
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: victoria-metrics-cluster
        app.kubernetes.io/version: 0.1.0-cluster
        helm.sh/chart: victoria-metrics-cluster-0.1.1
      name: RELEASE-NAME-victoria-metrics-cluster-vmselect
      namespace: NAMESPACE
    spec:
      rules:
        - host: vmselect.local
          http:
            paths:
              - backend:
                  service:
                    name: RELEASE-NAME-victoria-metrics-cluster-vmselect
                    port:
                      name: http
                path: /select
                pathType: Prefix
